### ISSUE TITLE
Allow raising PR to different branch

### ISSUE THREAD

**Issue by jembishop** [MarcoIeni/release-plz#1424]
## Motivations

I would quite like to use a git workflow where I maintain a separate release branch for my releases. This allows us to have our "main" branch be the latest that all developers are working on and the "release" branch be just the releases. The advantage of this is that hotfixes can be merged directly into "release" without having to bring over all the other changes in "main".

In this workflow using "release-plz", I first have to create a pull request to "release" and merge that in and then I have to merge another pull request that "release-plz" raises. It would be more convenient if release-plz could directly raise 1 PR when I invoke the CLI tool, which could be run by a manually invoked Github action.

Going further, maybe you have a staging branch and you want your flow to be to release from main to staging then staging to release. 

- Would you like to implement this feature? [y/n]

I could implement this feature, looking at the code it looks fairly straightforward. 

## Solution

Basically there would be another option config table option.

```toml

[branches.main]
target = "staging"

[branches.staging]
target = "release"

[branches."hotfix-.*"]
regex = true
target = "release"

```

When you run the `release-pr` command it will raise the pull request against the requested target branch instead of the current branch. The "from" branch can be a regex if the `regex = true` option is set.

## Alternatives
Alternative it is to continue with current workflow of merging 2 PRs.

Interested to know what you think. Thanks.

**Comment by MarcoIeni** [comment]
> I would quite like to use a git workflow where I maintain a separate release branch for my releases. This allows us to have our "main" branch be the latest that all developers are working on and the "release" branch be just the releases. The advantage of this is that hotfixes can be merged directly into "release" without having to bring over all the other changes in "main".

Ok, that's clear to me 👍
So the `release` branch is what's currently published on crates.io

> In this workflow using "release-plz", I first have to create a pull request to "release" and merge that in 

This is a PR to add the changes from `main` to the `release` branch? If not, what PR is this? I.e. what are the head and base branches?

> It would be more convenient if release-plz could directly raise 1 PR when I invoke the CLI tool

How would this PR look like? Can you specify the head and base branches?

> Going further, maybe you have a staging branch and you want your flow to be to release from main to staging then staging to release.

I'm not sure I want to implement this logic into release-plz, it sounds too specific to me.
However I want to add the basic building blocks in release-plz to allow you to implement this logic yourself, without manual steps.

> When you run the release-pr command it will raise the pull request against the requested target branch instead of the current branch

Ok, so you want to run `release-plz release-pr` against a different base branch (not `main`)?
If you want to use `release` as the base branch, you first need to add the changes from `main` to `release`, right?

I still didn't get how the `[branches]` configuration works.
From a first look my instinct tells me to not add this to release-plz.
But I'm keen to learn more about it. 👍

Maybe your workflow can be automated by using the `gh` CLI tool to create the PRs you need before running release-plz? 🤔

**Comment by jembishop** [comment]
Thanks for your reply.

I guess you are right in that I can create and existing CI job to automate this which automatically merges in main into release, which should trigger `release-plz`. That seems like it will be a better solution. Thanks for your advice!

**Comment by MarcoIeni** [comment]
It would be great to document your use case and how you solved it under [extra](https://release-plz.ieni.dev/docs/extra) 😁
If you feel like, PR welcome!
